packages/astro/src/core/ssr/rss.ts

```diff
@@ -47,9 +47,12 @@ export function generateRSS(args: GenerateRSSArgs): string {
 		if (typeof result !== 'object') throw new Error(`[${srcFile}] rss.items expected an object. got: "${JSON.stringify(result)}"`);
 		if (!result.title) throw new Error(`[${srcFile}] rss.items required "title" property is missing. got: "${JSON.stringify(result)}"`);
 		if (!result.link) throw new Error(`[${srcFile}] rss.items required "link" property is missing. got: "${JSON.stringify(result)}"`);
+		let itemLink = result.link;
+		// If the item's link is already a valid URL, don't mess with it.
+		if (!/^(?:https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&//=]*)$/i.test(result.link)) itemLink = canonicalURL(result.link, site).href;
```

Comment:
Complicated RegExps like this scare me from a maintenance perspective. 

Could this be simplified to use something like...
```
const isValidURL = (value) => {
  try {
    new URL(result.link)
    return true;
  } catch (e) {}
  return false;
}
```

Reply:
Quite possibly, but I wasn't sure which would be more performant.